quantum/process_keycode/process_tap_dance.h

```diff
@@ -56,20 +59,37 @@ typedef struct {
     void (*layer_function)(uint8_t);
 } tap_dance_dual_role_t;
 
-#    define ACTION_TAP_DANCE_DOUBLE(kc1, kc2) \
-        { .fn = {tap_dance_pair_on_each_tap, tap_dance_pair_finished, tap_dance_pair_reset}, .user_data = (void *)&((tap_dance_pair_t){kc1, kc2}), }
+#    ifndef TAP_DANCE_ON_EACH_RELEASE
```

Comment:
This is ... messy, at best. 

If anything, I'd say that this should just have the on_each_release option, and just have the ifdef in the function (as it is in the c file).

Reply:
Yea, I was thinking of the same approach initially, but I was trying to ensure that this feature didn't affect the memory usage for keyboards that didn't use that functionality, hence why I went with this implementation instead. 

But, as you mentioned, the implementation turned out less than ideal; in the newest commit, I removed the macro-conditionals such that the new function parameter is always present.  There's no need for the `TAP_DANCE_ON_EACH_RELEASE` anymore (since the function added in the `c` file will check for `NULL`).

Thanks for the initial review and comments, and let me know what you think of the changes.